modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/accept/language/AcceptLanguageImpl.java

```diff
@@ -112,6 +113,17 @@ public Locale getPreferredLocale() {
 		List<Locale> locales = getLocales();
 
 		if (ListUtil.isNotEmpty(locales)) {
+			if (locales.size() == 1) {
+				return locales.get(0);
+			}
+
+			String acceptLanguage = _httpServletRequest.getHeader(
+				HttpHeaders.ACCEPT_LANGUAGE);
+
+			if (LanguageUtil.isAvailableLanguageCode(acceptLanguage)) {
```

Comment:
You are assuming that `acceptLanguage` will be a single language, but it can actually be a complex priority list of locales like `w,en;q=0.7,en;q=0.3` in which case this line would not work.